### PR TITLE
Support Nauty digraph6 format

### PR DESCRIPTION
**Pull request by mtorpey**
As discovered in Issue #158, our version of digraph6 differs from that in Nauty, but the two formats cannot be confused with each other (since they start with `+` and `&` respectively).  This PR adds support for reading the Nauty format, while preserving the ability to read the old one (with an info warning).  This PR makes us write the Nauty format, and removes the ability to write the old one (we should use Nauty's format from now on).

### CODE REVIEW

File: gap/io.gi [hunk 1]
```diff
@@ -821,8 +826,14 @@ function(s)
...
 
-  # Check for the special '+' character
-  if s[1] <> '+' then
+  # Check for the special '&' character (or the deprecated '+')
+  if s[1] = '&' then
+    legacy := false;
+  elif s[1] = '+' then
+    legacy := true;
+    Info(InfoDigraphs, 1, "Digraph6 strings beginning with '+' are deprecated");
+    Info(InfoDigraphs, 1, "and may not be recognised by other programs.");
```
Comment (wilfwilson) [inline gap/io.gi]: I think this warning needs to be more forceful/explicit. You could read this and think that the only thing that changed was that the '+' has been changed to a '&'. But the actual encoding is fundamentally different.

Maybe something like

> Digraph6 strings that begin with '+' use an old specification of the Digraph6 format that is incompatible with the current, and now standard, specification. These old strings can still be read by the Digraphs package, but it is unlikely that they can be read correctly by other programs. We recommend that you re-encode your digraphs with the new format.

**Comment by mtorpey** [comment]
Altered as requested.  See what you think of the new info warning – succinct enough? ;)

**Comment by mtorpey** [comment]
Altered as requested. See what you think of the new info warning – succinct enough? :wink:

**Comment by james-d-mitchell** [comment]
Great, super thanks @mtorpey !!

**Pull request merged**